src/ripple/app/paths/RippleState.h

```diff
@@ -153,27 +151,27 @@ class RippleState
     getJson(int);
 
 private:
-    std::shared_ptr<SLE const> sle_;
-
-    bool mViewLowest;
+    uint256 key_;
 
-    std::uint32_t mFlags;
-
-    STAmount const& mLowLimit;
-    STAmount const& mHighLimit;
+    STAmount const mLowLimit;
+    STAmount const mHighLimit;
 
-    AccountID const& mLowID;
-    AccountID const& mHighID;
+    AccountID const mLowID;
+    AccountID const mHighID;
```

Comment:
You actually don't _need_ the `mLowID` and `mHighID` variables: you can always access those via `m[Low|High]Limit.getIssuer()` without incurring any significant runtime cost. This saves 40 more bytes.

Reply:
That's a good idea. Will do.

In the longer term, we should replace the `STAmount` variables with `IOUAmount` variables and keep the currency and issuers separate. The reason I don't want to do that now is we'd have to carefully audit those two classes to make sure all the corner cases are handed exactly the same.